src/main/java/com/rsr5/gocd/riemann_notifier/PipelineDetailsPopulator.java

```diff
@@ -54,32 +100,45 @@ public JsonObject extendMessage(String requestBody) {
             JsonObject pipelineObject = json.get("pipeline").getAsJsonObject();
             JsonObject stageObject = pipelineObject.get("stage").getAsJsonObject();
 
+            if ("Building".equals(stageObject.get("state").getAsString())) {
+                return json;
+            }
+
             String pipeline = pipelineObject.get("name").getAsString();
             String pipelineCounter = pipelineObject.get("counter")
                     .getAsString();
             String stage = stageObject.get("name").getAsString();
             String stageCounter = stageObject.get("counter").getAsString();
 
-            JsonElement extraDetails = downloadPipelineDetails(pipeline);
-            json = mergeInPipelineInstanceDetails("x-pipeline-instance" +
-                    "-details", json, extraDetails);
-
             json.add("x-pipeline-artifacts", new JsonObject());
             JsonArray jobArtifacts = new JsonArray();
 
             JsonArray jobs = stageObject.get("jobs").getAsJsonArray();
             for(final JsonElement job : jobs) {
-
-                extraDetails = downloadPipelineArtifacts(pipeline, stage,
-                        pipelineCounter, stageCounter, job.getAsJsonObject().get("name").getAsString());
-                jobArtifacts.add(extraDetails);
+                String jobName = job.getAsJsonObject().get("name").getAsString();
+
+                JsonElement artifacts = downloadPipelineArtifacts(pipeline, stage, pipelineCounter, stageCounter, jobName);
+                jobArtifacts.add(artifacts);
+
+                if (null != this.config.getJUnitJobWhitelist() &&
+                    Pattern.matches(this.config.getJUnitJobWhitelist(), jobName)) {
+                    try {
+                        JsonElement junitResults = findJunitResults(artifacts.getAsJsonObject().get(jobName).getAsJsonArray());
+                        if (null != junitResults) {
+                            json.add("x-junit-results", junitResults);
```

Comment:
🤔 why the `x-` prefixes here?

Reply:
This was done to be consistent with what the original plugin author had done.